common/network_manager.go

```diff
@@ -593,37 +628,39 @@ func (m *NetworkManager) updateProtocolRequestInterception() error {
 		return nil
 	}
 	m.protocolReqInterceptionEnabled = enabled
-	if enabled {
-		actions := []Action{
-			network.SetCacheDisabled(true),
-			fetch.Enable().
-				WithHandleAuthRequests(true).
-				WithPatterns([]*fetch.RequestPattern{
-					{URLPattern: "*"},
-				}),
-		}
-		for _, action := range actions {
-			if err := action.Do(cdp.WithExecutor(m.ctx, m.session)); err != nil {
-				return fmt.Errorf("unable to execute %T: %w", action, err)
-			}
-		}
-	} else {
-		actions := []Action{
+
+	actions := []Action{
+		network.SetCacheDisabled(true),
+		fetch.Enable().
+			WithHandleAuthRequests(true).
+			WithPatterns([]*fetch.RequestPattern{
+				{
+					URLPattern:   "*",
+					RequestStage: fetch.RequestStageRequest,
+				},
+			}),
+	}
+	if !enabled {
+		actions = []Action{
 			network.SetCacheDisabled(false),
 			fetch.Disable(),
 		}
-		for _, action := range actions {
-			if err := action.Do(cdp.WithExecutor(m.ctx, m.session)); err != nil {
-				return fmt.Errorf("unable to execute %T: %w", action, err)
-			}
+	}
+	for _, action := range actions {
+		if err := action.Do(cdp.WithExecutor(m.ctx, m.session)); err != nil {
+			return fmt.Errorf("cannot execute %T: %w", action, err)
 		}
 	}
+
 	return nil
 }
 
 // Authenticate sets HTTP authentication credentials to use.
 func (m *NetworkManager) Authenticate(credentials *Credentials) {
 	m.credentials = credentials
+	if credentials != nil {
+		m.userReqInterceptionEnabled = true
```

Comment:
It feels out of place for `Authenticate()` to also be setting this, but OK, not a big deal.

Reply:
Sure!

Reply:
I'll create a refactoring issue about this because I have different views.